compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -865,8 +865,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     (lhs_expr.span, msg, suggestion)
                 } else {
                     let msg = format!("{} and panic if the converted value doesn't fit", msg);
-                    let suggestion =
-                        format!("{}{}.try_into().unwrap()", prefix, with_opt_paren(&src));
+                    // #71309: `src` may contain `?` at the end, trim it
+                    // as it makes suggestions incorrect.
+                    let suggestion = format!(
+                        "{}{}.try_into().unwrap()",
+                        prefix,
+                        with_opt_paren(&src).trim_end_matches('?')
+                    );
                     (expr.span, msg, suggestion)
```

Comment:
Ah! I see. That `trim_end_matches` is incorrect. The case I mentioned is [supposed to suggest `x?.try_into().unwrap()`](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=38ea847d49b9cb2c1e7f9a6d27481534), but this "blind" removal of the trailing `?` is getting rid of it. Do we have a test where the removal of the `?` is correct?

Reply:
Oh, somehow I misunderstood your previous review comment (https://github.com/rust-lang/rust/pull/75029#discussion_r483332576), fixed.